### PR TITLE
run integration tests in kind

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -21,29 +21,25 @@
...
+  version = "~0.1.0"
 
-[[constraint]]
-  branch = "master"
+[[override]]
   name = "github.com/giantswarm/backoff"
-
-[[constraint]]
-  branch = "master"
-  name = "github.com/giantswarm/e2e-harness"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Getting rid of `e2e-harness` finally. 🎉

File: controller/collector/timestamp.go [hunk 2]
```diff
@@ -99,6 +99,7 @@ func (t *Timestamp) Collect(ch chan<- prometheus.Metric) error {
 			)
 		}
 	}
+
```
Comment (xh3b4sd) [inline controller/collector/timestamp.go]: 🙈

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,113 +1,72 @@
...
-
     - run: ./architect build
-
     - store_test_results:
         path: /tmp/results
-
-    - deploy:
-        command: |
-          if [ "${CIRCLE_BRANCH}" == "master" ]; then
-            ./architect deploy
```
Comment (xh3b4sd) [inline .circleci/config.yml]: I feel like this deployment thingy here did not make sense at all.

File: README.md [hunk 4]
```diff
@@ -64,6 +64,38 @@ operators][operators].
...
+
+```
+kind create cluster
+```
+
+The tests need to figure out how to connect to the Kubernetes cluster. Therefore
+we need to set an environment variable pointing to your local kube config.
+
+```
+export E2E_KUBECONFIG=<your-path>/.kube/config
```
Comment (rossf7) [inline README.md]:
```suggestion
export E2E_KUBECONFIG=~/.kube/config
```

This should work for MacOS and Linux in most situations. If you're using Windows you've got bigger problems. :trollface:

**Comment by xh3b4sd** [comment]
It happens once in a while that some tests flap. See https://circleci.com/gh/giantswarm/operatorkit/6782. 

```
#!/bin/bash -eo pipefail
kind create cluster --config kind-config --image quay.io/giantswarm/kind-node:v1.16.3
Creating cluster "kind" ...
 ✓ Ensuring node image (quay.io/giantswarm/kind-node:v1.16.3) 🖼
 ✓ Preparing nodes 📦 
 ✓ Writing configuration 📜
 ✗ Starting control-plane 🕹️
ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged kind-control-plane kubeadm init --ignore-preflight-errors=all --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 126

Exited with code exit status 1
```

@rossf7 you got an idea why that is?

**Comment by rossf7** [comment]
@xh3b4sd No I've also seen this flapping but not sure of the cause.

We're running the latest kind release `0.7.0` and I don't see any related open issues there. 
https://github.com/kubernetes-sigs/kind/releases/tag/v0.7.0

Could be a docker issue with the circle VM I guess?

Maybe we could retry in the integration-test job? That might also show if it's a temp issue or the VM is borked.

**Comment by xh3b4sd** [comment]
+1 on the retries in the job. I am just not sure where or how to do that. I would not let this block the PR here then. 

Another thing I noticed is that the image (whatever kind of image that is) is loaded all the time and we may want to check how to cache this. Just food for thought to make the setup even faster.

**Comment by xh3b4sd** [comment]
And let's maybe create an upstream issue.

**Comment by rossf7** [comment]
> +1 on the retries in the job. I am just not sure where or how to do that. I would not let this block the PR here then.

Yes it flaps enough to be annoying but not enough to block IMO.

I'm going to create a board for #wg-integration-testing tomorrow. I'll make sure we have cards / issues for this. Good point about checking with upstream. I'll start with k8s slack.

File: controller/integration/env/common.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package env
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	EnvVarE2EKubeconfig = "E2E_KUBECONFIG"
```
Comment (kopiczko) [inline controller/integration/env/common.go]: IMO this should be `INTEGRATION_KUBECONFIG`. We agreed to not call those tests e2e anymore.

Comment (rossf7) [reply controller/integration/env/common.go]: In this context I think the shorter name is useful. The same way we use G8sClient but Giantnetes is not a thing.

I'd be OK with KIND_KUBECONFIG but then we need to change it if we move to something else later.

Comment (kopiczko) [reply controller/integration/env/common.go]: `I8N_KUBECONFIG` :D Don't know.

Comment (fiunchinho) [reply controller/integration/env/common.go]: It's in the `env` package under the `integration` folder, maybe `E2E` or `I8N` are redundant. It's the tenant cluster kubeconfig, right? What about `TC_KUBECONFIG`?

Comment (kopiczko) [reply controller/integration/env/common.go]: https://gigantic.slack.com/archives/CUTGJ4NKA/p1584529656034000

**Pull request merged**